src/test/e2e/api/client/feature.token.deleted.project.e2e.test.ts

```diff
@@ -0,0 +1,153 @@
+import { type IUnleashTest, setupAppWithAuth } from '../../helpers/test-helper';
+import dbInit, { type ITestDb } from '../../helpers/database-init';
+import getLogger from '../../../fixtures/no-logger';
+import type { ApiTokenService } from '../../../../lib/services/api-token-service';
+import { ApiTokenType } from '../../../../lib/types/models/api-token';
+import { DEFAULT_ENV } from '../../../../lib/util/constants';
+import { TEST_AUDIT_USER } from '../../../../lib/types';
+import { User } from '../../../../lib/server-impl';
+
+let app: IUnleashTest;
+let db: ITestDb;
+
+let apiTokenService: ApiTokenService;
+
+const environment = 'testing';
+const project = 'default';
+const project2 = 'some';
+const deletionProject = 'deletion';
+const deletionTokenName = 'delete';
+const feature1 = 'f1.token.access';
+const feature2 = 'f2.token.access';
+const feature3 = 'f3.p2.token.access';
+
+beforeAll(async () => {
+    db = await dbInit('feature_api_api_access_client_deletion', getLogger);
+    app = await setupAppWithAuth(db.stores, {}, db.rawDatabase);
+    apiTokenService = app.services.apiTokenService;
+
+    const { featureToggleServiceV2, environmentService } = app.services;
+    const { environmentStore, projectStore } = db.stores;
+
+    await environmentStore.create({
+        name: environment,
+        type: 'test',
+    });
+
+    await projectStore.create({
+        id: project2,
+        name: 'Test Project 2',
+        description: '',
+        mode: 'open' as const,
+    });
+
+    await projectStore.create({
+        id: deletionProject,
+        name: 'Deletion Project',
+        description: '',
+        mode: 'open' as const,
+    });
+
+    await environmentService.addEnvironmentToProject(
+        environment,
+        project,
+        TEST_AUDIT_USER,
+    );
+    await environmentService.addEnvironmentToProject(
+        environment,
+        project2,
+        TEST_AUDIT_USER,
+    );
+
+    await featureToggleServiceV2.createFeatureToggle(
+        project,
+        {
+            name: feature1,
+            description: 'the #1 feature',
+        },
+        TEST_AUDIT_USER,
+    );
+
+    await featureToggleServiceV2.createStrategy(
+        {
+            name: 'default',
+            constraints: [],
+            parameters: {},
+        },
+        { projectId: project, featureName: feature1, environment: DEFAULT_ENV },
+        TEST_AUDIT_USER,
+    );
+    await featureToggleServiceV2.createStrategy(
+        {
+            name: 'flexibleRollout',
```

Comment:
```suggestion
            name: 'default',
```

Flexible rollout with no parameters is definitely broken. I know it doesn't matter for this PR, but this PR was complex because a test was naughty about a data constraint, so let's just be good to ourselves in future and pick a strategy that's valid without parameters